crates/rethnet_evm_napi/src/state.rs

```diff
@@ -81,10 +82,38 @@ fn add_precompiles(accounts: &mut HashMap<Address, AccountInfo>) {
 #[napi(custom_finalize)]
 #[derive(Debug)]
 pub struct StateManager {
-    pub(super) state: Arc<RwLock<dyn SyncState<StateError>>>,
+    state: Arc<RwLock<dyn SyncState<StateError>>>,
     context: Arc<Context>,
 }
 
+impl StateManager {
+    /// Returns the instance's [`Context`].
+    pub fn context(&self) -> &Context {
+        &self.context
+    }
+
+    #[cfg_attr(feature = "tracing", tracing::instrument(skip_all))]
+    fn with_state<S>(env: &mut Env, context: &RethnetContext, state: S) -> napi::Result<Self>
+    where
+        S: SyncState<StateError>,
+    {
+        env.adjust_external_memory(STATE_MEMORY_SIZE)?;
```

Comment:
Could be good to have a comment explaining this line

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/4175/commits/eb051b809524a1ac4c25f10767dca08fd0ec6d7c